### PR TITLE
custom op full_like

### PR DESCRIPTION
**Pull request by Alicia1529**
## Description ##
Add numpy operator full_like

### CODE REVIEW

File: python/mxnet/numpy_op_fallback.py [hunk 1]
```diff
@@ -49,6 +49,73 @@ def _register_helper(prop_cls):
...
+        self._fill_value = fill_value
+        self._dtype = dtype
+        self._order = order 
+        self._subok = subok
+        self._shape = shape
+
+    def forward(self, is_train, req, in_data, out_data, aux):
+        np_version = np.version.version
+        if np_version == '1.17.0':
+            out = np.full_like(in_data[0].asnumpy(), self._fill_value, dtype=self._dtype, order=self._order,
```
Comment (wkcn) [inline python/mxnet/numpy_op_fallback.py]: It will be faster to use `mx.nd.full(shape, fill_value)`

**Comment by hgt312** [comment]
@mxnet-label-bot add [Numpy]

File: tests/python/unittest/test_numpy_op.py [hunk 2]
```diff
@@ -4463,6 +4464,52 @@ def hybrid_forward(self, F, a):
...
+    class TestFullLike(HybridBlock):
+        def __init__(self, fill_value, dtype):
+            super(TestFullLike, self).__init__()
+            self._fill_value = fill_value
+            self._dtype = dtype
+
+        def hybrid_forward(self, F, x, *args, **kwargs):
+            return F.np.full_like(x, self._fill_value, self._dtype)
+
+    if StrictVersion(platform.python_version()) < StrictVersion('3.0.0'):
```
Comment (reminisce) [inline tests/python/unittest/test_numpy_op.py]: You can decorate the test function with this condition as the following.
```python
@unittest.skipIf(StrictVersion(platform.python_version()) < StrictVersion('3.0.0'), "reason to skip")
def test_np_full_like():
    ...
```

Comment (Alicia1529) [reply tests/python/unittest/test_numpy_op.py]: thanks!

**Comment by sxjscience** [comment]
I suggest not use the customop for full_like, which is very similar to ones_like

**Comment by sxjscience** [comment]
@Alicia1529, You can refer to the implementation of ones_like and zeros_like: https://github.com/apache/incubator-mxnet/blob/61c8bafdcfee129e4f7a491438a2402e6762ddd9/src/operator/numpy/np_init_op.cc#L100-L134

**Comment by Alicia1529** [comment]
> @Alicia1529, You can refer to the implementation of ones_like and zeros_like:
> 
> https://github.com/apache/incubator-mxnet/blob/61c8bafdcfee129e4f7a491438a2402e6762ddd9/src/operator/numpy/np_init_op.cc#L100-L134

Thanks for the suggestions!

In pr #16804 a c++ version of full_like is implemented. Zeros_like and ones_like are also rewritten by calling full_like with different fill_value in the front-end to fix the type inference bug in issue #16646 

But I am still fixing some problems in python2 test, so it has not been merged. This pr is older and should be closed.

**Comment by sxjscience** [comment]
@Alicia1529 Is it safe to close this issue?

**Comment by Alicia1529** [comment]
@sxjscience  Yeah, thanks

**Pull request closed without merging**